### PR TITLE
Cleanup for multi ip address

### PR DESCRIPTION
**Pull request by ritwickdsouza**
References #59 
I will work on this as i get feedback from the community. Right now it generates the desired output.
All test cases pass except `test_multiple_ip (tests.openwrt.test_network.TestNetworkRenderer)` which needs to be changed.
Let me know if any other changes are required.

### CODE REVIEW

**Comment by ritwickdsouza** [comment]
@nemesisdesign How should i go ahead with this PR ?

**Comment by ritwickdsouza** [comment]
@nemesisdesign Since I'm done with my exams for sometime now, I would love to work on this. How should i go ahead ?

**Comment by nemesisdesign** [comment]
@ritwickdsouza you should try to do some refactoring first because the current `_get_interfaces` method is becoming too big and hard to maintain. Can you split it into separate dedicated methods?

Maybe you can split the work to solve this issue in 2 phases:

1. refactor the current implementation of `_get_interfaces`: make it easier to read, consider delegating some tasks to new methods
2. create a new method that only deals with generating the right output to solve this issue

PS: pay attention to travis build, if it fails for some trivial reason like it is failing now for your pull request, always fix it, see https://travis-ci.org/openwisp/netjsonconfig/jobs/218063069

**Comment by ritwickdsouza** [comment]
Working on it now.

**Comment by ritwickdsouza** [comment]
Created a function as suggested for this issue.

File: netjsonconfig/backends/openwrt/renderers.py [hunk 1]
```diff
@@ -158,6 +160,18 @@ def _get_interfaces(self):
                 counter += 1
         return uci_interfaces
 
+    def __multi_ipaddr(self, address_list):
```
Comment (nemesisdesign) [inline netjsonconfig/backends/openwrt/renderers.py]: I cannot understand this method very much.

It's better we start again from scratch.

File: netjsonconfig/backends/openwrt/renderers.py [hunk 2]
```diff
@@ -158,6 +160,18 @@ def _get_interfaces(self):
...
+        for i in address_list:
```
Comment (nemesisdesign) [inline netjsonconfig/backends/openwrt/renderers.py]: `i` not very good variable name

File: netjsonconfig/backends/openwrt/renderers.py [hunk 3]
```diff
@@ -158,6 +160,18 @@ def _get_interfaces(self):
...
+                for j in address_list:
```
Comment (nemesisdesign) [inline netjsonconfig/backends/openwrt/renderers.py]: `j` also not a good variable name

File: netjsonconfig/backends/openwrt/renderers.py [hunk 4]
```diff
@@ -158,6 +160,18 @@ def _get_interfaces(self):
...
+                    if i.get('family') == 'ipv4' and j.get('family') == 'ipv6':
```
Comment (nemesisdesign) [inline netjsonconfig/backends/openwrt/renderers.py]: indentation is 8 while it should be 4

File: netjsonconfig/backends/openwrt/renderers.py [hunk 5]
```diff
@@ -158,6 +160,18 @@ def _get_interfaces(self):
...
+                        if j.get('address') and j.get('mask'):
+                            address_value = '{address}/{mask}'.format(**j)
+                            i['ip6addr'] = address_value
+                            temp = address_list.index(j)
+                if temp:
```
Comment (nemesisdesign) [inline netjsonconfig/backends/openwrt/renderers.py]: `temp` not always defined and causes many tests to fail

**Comment by nemesisdesign** [comment]
testing gitter integration

**Pull request closed without merging**